ArduSub/AP_Arming_Sub.cpp

```diff
@@ -53,6 +53,37 @@ bool AP_Arming_Sub::arm(AP_Arming::Method method, bool do_arming_checks)
         return false;
     }
 
+    // if there's no shift assigned, it must not be a shift funtion
```

Comment:
This should be a pre-arm check, not a "it fails when you go to arm it" thing.

Reply:
Pre-arm checks may be disabled. I think this check should not be allowed to be disabled. Is that achievable with the AP_Arming library?

Reply:
So then, what is the verdict. Should I change this into a pre-arm check that cannot be disabled?

Also, should i solve each of these conversations with separate commits onto this branch? (sorry, bit of a noob question)

Reply:
You can push new commits directly to this branch to address the review. We will ask you to fixup/squash them before merge, or you may do that as you go while you update the branch during the review.

Reply:
> So then, what is the verdict. Should I change this into a pre-arm check that cannot be disabled?

Yes, according to @peterbarker's recommendation.